presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -266,13 +267,18 @@ public PlanWithProperties visitAggregation(AggregationNode node, StreamPreferred
                 return rebaseAndDeriveProperties(node, ImmutableList.of(child));
             }
 
-            // if aggregation is decomposable and not already parallel, push down a partial which will be executed in parallel
+            // If the following conditions satisfy, push down a partial which will be executed in parallel.
+            // * aggregation is decomposable, and
+            // * aggregation is not already decomposed, and
+            // * aggregation will not run in parallel unless a partial aggregation is pushed down
```

Comment:
I'm not sure what this one means... the in the structure of the comment you seem to be listing conditions, but this last one doesn't seem like a condition and is a comment instead